shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala

```diff
@@ -0,0 +1,108 @@
+package amf.plugins.document.vocabularies.emitters.instances
+import amf.core.annotations.LexicalInformation
+import amf.core.emitter.{EntryEmitter, RenderOptions, SpecOrdering}
+import amf.core.parser.Position
+import amf.core.parser.Position.ZERO
+import amf.plugins.document.vocabularies.model.document.{Dialect, DialectInstanceUnit}
+import amf.plugins.document.vocabularies.model.domain.{DialectDomainElement, NodeMappable, PublicNodeMapping, UnionNodeMapping}
+import org.yaml.model.YDocument.EntryBuilder
+import org.yaml.model.YNode
+import amf.core.utils.AmfStrings
+
+case class DeclarationsGroupEmitter(declared: Seq[DialectDomainElement],
+                                    publicNodeMapping: PublicNodeMapping,
+                                    nodeMappable: NodeMappable,
+                                    instance: DialectInstanceUnit,
+                                    dialect: Dialect,
+                                    ordering: SpecOrdering,
+                                    declarationsPath: Seq[String],
+                                    aliases: Map[String, (String, String)],
+                                    keyPropertyId: Option[String] = None,
+                                    renderOptions: RenderOptions)
+    extends EntryEmitter
+    with DialectEmitterHelper {
+
+  def computeIdentifier(decl: DialectDomainElement): String = {
+    decl.declarationName.option() match {
+      case Some(name) => name
+      case _ =>
+        decl.id
+          .split("#")
+          .last
+          .split("/")
+          .last
+          .urlDecoded // we are using the last part of the URL as the identifier in dialects
+    }
+  }
+
+  override def emit(b: EntryBuilder): Unit = {
+    val discriminator = findNodeMappingById(publicNodeMapping.mappedNode().value()) match {
+      case (_, unionMapping: UnionNodeMapping) =>
+        Some(DiscriminatorHelper(unionMapping, this))
+      case _ => None
+    }
+
+    if (declarationsPath.isEmpty) {
+      val declarationKey = publicNodeMapping.name().value()
+      b.entry(
+          declarationKey,
+          _.obj { b =>
+            sortedDeclarations().foreach { decl =>
+              val identifier = computeIdentifier(decl)
+              b.entry(
```

Comment:
If we extract the entry emittion to another emitter of a NamedDialectDomainElement we can map the declaration to emitter and do the traverse+sorted as amf wbapi emitter does.
Also, that will be like a first step to create a declared element isolated emiter.

Reply:
Good suggestion, keeping it in mind for further refactors.